src/inference/gibbs_conditional.jl

```diff
@@ -49,7 +49,7 @@ m = inverse_gdemo(x)
 sample(m, Gibbs(GibbsConditional(:λ, cond_λ), GibbsConditional(:m, cond_m)), 10)
 ```
 """
-struct GibbsConditional{S, C}
+struct GibbsConditional{S, C} <: InferenceAlgorithm
```

Comment:
Can this change be reverted
```suggestion
struct GibbsConditional{S, C}
```
and instead the type restrictions of the Gibbs sampler and its constructors above be relaxed (`isgibbscomponent` in the inner constructor should be sufficient, I assume)? The problem is that any InferenceAlgorithm can be used in the `sample` calls but this really is not a standalone sampler.